### PR TITLE
[issue-147] Encapsulate the steps to run a CLI operations into a simple function that can be invoked sourcing install-common.sh

### LINKED ISSUE #147: Enable an API to allow the execution of CLI operations at S2I phase
We can make easier for the users to add CLI operations at S2I phase removing any need to start manually the embedded server.

The intention would be at `install.sh`. Currently, a user who is sourcing the `/usr/local/s2i/install-common.sh` is able to use the following common functions 

1.  install_deployments
2. install_modules
3. configure_drivers

The idea is to add a new one, run_cli_script that receives by argument a file initialized by the user that contains all the operations added by the user to configure the server. Basically, users could do the following in the install.sh file:

```
echo "Injected dir is $1"
S2I_CLI_SCRIPT=$1/my_script.cli

echo "/system-property=foo:add(value=bar)" >> "${S2I_CLI_SCRIPT}"
run_cli_script "${S2I_CLI_SCRIPT}"
```

run_cli_script will start the embedded server and the content of ${S2I_CLI_SCRIPT} will be executed.

We have to keep in mind that `configure_drivers` is likely to start the embedded server to configure any drivers. If the user is using both functions in the install.sh, we will be executing the embedded server twice, but being on S2I phase, I guess this would be acceptable

For the runtime phase, we do not need to change anything since the users already can use the extensions mechanism by using `preconfigure.sh` and `postconfigure.sh` scripts to append operations to "${CLI_SCRIPT_FILE}". (We plan to blog about this after getting WF19 out)

If we agreed with the proposal, we can implement this enablement for the S2I phase in WF19.

### PR DESCRIPTION
**Pull request by yersan**
Fixes #147

### CODE REVIEW

File: jboss/container/wildfly/s2i/bash/artifacts/opt/jboss/container/wildfly/s2i/install-common/install-common.sh [hunk 1]
```diff
@@ -120,3 +120,19 @@ function configure_drivers(){
   )
 }
 
+function run_cli_script(){
+  if [ $# != 1 ]; then
+    echo "Usage: Script file parameter required."
```
Comment (jfdenise) [inline jboss/container/wildfly/s2i/bash/artifacts/opt/jboss/container/wildfly/s2i/install-common/install-common.sh]: Should we use log_error? I am wandering if we should exit 1? I see that other functions just print the message and return. We are possibly hiding a user mis-configuration that could then be difficult to identify at runtime.

Comment (yersan) [reply jboss/container/wildfly/s2i/bash/artifacts/opt/jboss/container/wildfly/s2i/install-common/install-common.sh]: I asked myself the same question, but the other methods exposed in the same file use also a simple echo. So finally, I wanted to be consistent.

I'm not sure if we can change all, just only the new one. We could use a warning instead to make the trace more relevant.

WDYT?

Comment (jfdenise) [reply jboss/container/wildfly/s2i/bash/artifacts/opt/jboss/container/wildfly/s2i/install-common/install-common.sh]: @yersan , I would not touche the existing one. This is perhaps just something we would need to document for the API you introduce. You could add a chapter in WildFly s2i README.md

**Comment by yersan** [comment]
Closing by now; we need to move as a new feature only in wildfly-s2i

**Pull request closed without merging**